test/test_local_io.py

```diff
@@ -317,6 +317,23 @@ def is_nonempty_json(path_and_stream):
         ]
         self.assertEqual(expected_res, list(json_dp))
 
+        def as_dict(data):
+            return {"file_name": data[0], "stream": data[1]}
+
+        dict_dp = datapipe_nonempty.map(as_dict)
+        parsed_dp = dict_dp.parse_json_files(input_col="stream", output_col="json")
```

Comment:
This is the most important part of the PR. IMO it will simplify some of the domains code.